src/pages/collections/index.tsx

```diff
@@ -1,13 +1,111 @@
-import Head from 'next/head';
+import styled from 'styled-components';
+import { useState, useEffect } from 'react';
+
 import { getServerSideUserProps } from '@utils';
+import { useQuery } from '@tanstack/react-query';
+import { KEY } from '@static';
+import { getCollection } from '@api';
+import { Loading, Card, Typography } from '@components';
+import Link from 'next/link';
+import { Collection } from '@types';
+import { NavLayout } from '@layouts';
+import { getFilteredCardList } from '@utils';
+
+const CollectionContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  gap: 16px;
+  max-height: 100vh;
+`;
+
+const FoundWrapper = styled.div`
+  display: flex;
+  margin-bottom: 24px;
+`;
+
+const NotFoundWrapper = styled.div`
+  text-align: center;
+  padding: 90% 0;
+`;
 
 export default function CollectionsPage() {
+  const [searchText, setSearchText] = useState('');
+  const [filteredCards, setFilteredCards] = useState<Collection[]>([]);
+  const [showText, setShowText] = useState(false);
+
+  const { data, isError, isLoading } = useQuery({
+    queryKey: [KEY.COLLECTION],
+    queryFn: getCollection,
+  });
+
+  useEffect(() => {
+    const filteredData = getFilteredCardList(searchText, data?.data || []);
+    setFilteredCards(filteredData);
+  }, [searchText, data]);
+
   return (
     <>
-      <Head>
-        <title>Create Next App</title>
-      </Head>
-      <div>보관함 페이지</div>
+      {isError ? (
+        <div>ERROR</div>
+      ) : isLoading ? (
+        <Loading />
+      ) : (
+        <>
+          <NavLayout
+            searchText={searchText}
+            onSearchTextChange={setSearchText}
+            onShowTextChange={setShowText}
+          />
+          {showText && filteredCards.length > 0 && (
+            <FoundWrapper>
+              <Typography type="title2" systemColor="main">
+                {filteredCards.length}
+              </Typography>
+              <Typography type="title2" grayColor="white">
+                개의 명함을 찾았어요!
+              </Typography>
+            </FoundWrapper>
+          )}
+          <CollectionContainer>
+            {searchText ? (
+              filteredCards.length > 0 ? (
+                filteredCards.map((card) => (
+                  <Link href={`/cards/[id]`} as={`/cards/${card.cardId}`} key={card.encodeId}>
```

Comment:
```suggestion
                  <Link href={`/cards/${card.cardId}`} key={card.encodeId}>
```
최신 next에서는 as 필요없었던것같은데 기능상에 문제가 있나요?

Reply:
as 빼도 잘 작동하네요..! 수정했습니다!